Plugins/ExaTrkX/include/Acts/Plugins/ExaTrkX/Pipeline.hpp

```diff
@@ -0,0 +1,50 @@
+// This file is part of the Acts project.
+//
+// Copyright (C) 2023 CERN for the benefit of the Acts project
+//
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at http://mozilla.org/MPL/2.0/.
+
+#pragma once
+
+#include "Acts/Plugins/ExaTrkX/Stages.hpp"
+#include "Acts/Utilities/Logger.hpp"
+
+#include <any>
+#include <functional>
+#include <memory>
+#include <vector>
+
+#include <boost/multi_array.hpp>
+
+namespace Acts {
+
+class PipelineHook {
+ public:
+  virtual ~PipelineHook(){};
+  virtual void operator()(const std::any &, const std::any &) const {};
```

Comment:
I'm assuming these two arguments are always `nodes` and `edges`? Can you add thos names here?
Also this is not a pure-virtual class, is this on purpose? Otherwise consider making both of these pure-virtual.

Reply:
yeah, the non-purely-`virtual` method is on purpose to allow defaulting the argument in `Pipeline::run` like this: 
```c++
std::vector<std::vector<int>> run(std::vector<float> &features,
                                  std::vector<int> &spacepointIDs,
                                  const PipelineHook &hook = {}) const;
``` 
Other possibility would be to pass as pointer and check for nullptr... then it could be purely virtual...
Would you prefer such a solution?